src/block_disagg/block_disagg_ckpt.c

```diff
@@ -176,76 +149,21 @@ __wt_block_disagg_checkpoint_resolve(WT_BM *bm, WT_SESSION_IMPL *session, bool f
     } else {
         /* Keep all metadata for regular tables. */
         WT_SAVE_DHANDLE(
-          session, ret = __block_disagg_update_shared_metadata(bm, session, md_key, md_value));
+          session, ret = __wt_disagg_update_shared_metadata(session, md_key, md_value));
         WT_ERR(ret);
 
         /* Check if we need to include any other metadata keys. */
         if (WT_SUFFIX_MATCH(block_disagg->name, ".wt")) {
-            /* TODO: Change this hack to a nicer way of finding related metadata. */
-
-            WT_ERR(__wt_snprintf(md_key, len, "colgroup:%s", block_disagg->name));
+            WT_ERR(__wt_snprintf(md_key, len, "%s", block_disagg->name));
             md_key[strlen(md_key) - 3] = '\0'; /* Remove the .wt suffix */
-            md_cursor->set_key(md_cursor, md_key);
-            WT_ERR_NOTFOUND_OK(md_cursor->search(md_cursor), true);
-            if (ret == 0) {
-                WT_ERR(md_cursor->get_value(md_cursor, &md_value));
-                WT_SAVE_DHANDLE(session,
-                  ret = __block_disagg_update_shared_metadata(bm, session, md_key, md_value));
-                WT_ERR(ret);
-            }
-
-            WT_ERR(__wt_snprintf(md_key, len, "table:%s", block_disagg->name));
-            md_key[strlen(md_key) - 3] = '\0'; /* Remove the .wt suffix */
-            md_cursor->set_key(md_cursor, md_key);
-            WT_ERR_NOTFOUND_OK(md_cursor->search(md_cursor), true);
-            if (ret == 0) {
-                WT_ERR(md_cursor->get_value(md_cursor, &md_value));
-                WT_SAVE_DHANDLE(session,
-                  ret = __block_disagg_update_shared_metadata(bm, session, md_key, md_value));
-                WT_ERR(ret);
-            }
-
-            ret = 0; /* In case this is still set to WT_NOTFOUND from the previous step. */
+            WT_ERR_NOTFOUND_OK(__wt_disagg_copy_shared_metadata_layered(session, md_key), false);
         }
 
         /* Check if we need to include any other metadata keys for layered tables. */
         if (WT_SUFFIX_MATCH(block_disagg->name, ".wt_stable")) {
-            /* TODO: Change this hack to a nicer way of finding related metadata. */
-
-            WT_ERR(__wt_snprintf(md_key, len, "colgroup:%s", block_disagg->name));
```

Comment:
My current PR also modifies this code - I'll make an adjustment after you merge.